### PR TITLE
feat!(packer-images):Adding all Amazon builder logic for EC2 instances

### PR DESCRIPTION
**Pull request by jayfranco999**
Related to: https://github.com/jenkins-infra/helpdesk/issues/4316

draft PR (DO NOT MERGE)!!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,4 @@ manifest.json
 creds.json
 vars.json
 
+AWSCLIV2.pkg
```
Comment (jayfranco999) [inline .gitignore:14]:
```suggestion
AWSCLIV2.pkg

```

File: datasources.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+data "amazon-ami" "windows-2022" {
+  filters = {
+    name                = "Windows_Server-2022-English-Core-Base-*"
+    root-device-type    = "ebs"
+    virtualization-type = "hvm"
+  }
+  most_recent = true
+  owners      = ["amazon"]
+  region      = var.aws_region
+}
```
Comment (jayfranco999) [inline datasources.pkr.hcl:34]:
```suggestion
}

```

File: updatecli/updatecli.d/packer-plugins/amazon.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+actions:
+  default:
+    kind: github/pullrequest
+    title: Bump packer amazon plugin version to {{ source "lastReleaseVersion" }}
+    scmid: default
+    spec:
+      labels:
+        - dependencies
+        - packer-amazon-plugin
```
Comment (jayfranco999) [inline updatecli/updatecli.d/packer-plugins/amazon.yaml:50]:
```suggestion
        - packer-amazon-plugin
        
```

File: updatecli/updatecli.d/packer-plugins/amazon.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
```
Comment (jayfranco999) [inline updatecli/updatecli.d/packer-plugins/amazon.yaml:51]:
```suggestion
```

File: updatecli/updatecli.d/packer-plugins/amazon.yaml (hunk 3 above)
Comment (jayfranco999) [inline updatecli/updatecli.d/packer-plugins/amazon.yaml:50]:
```suggestion
        - packer-amazon-plugin
        
```

File: variables.pkr.hcl [hunk 5]
```diff
@@ -35,7 +48,7 @@ variable "image_version" {
 }
 variable "image_type" {
   type        = string
-  description = "Which kind of Packer builder to use (e.g. cloud platform): [azure-arm (default), docker]"
+  description = "Which kind of Packer builder to use (e.g. cloud platform): [amazon-ebs (default), azure-arm, docker]"
```
Comment (smerle33) [inline variables.pkr.hcl:51]:
```suggestion
  description = "Which kind of Packer builder to use (e.g. cloud platform): [amazon-ebs, azure-arm (default), docker]"
```

File: sources.pkr.hcl [hunk 6]
```diff
@@ -1,3 +1,44 @@
+# This source defines all the common settings for any AWS AMI (whatever Operating System)
+source "amazon-ebs" "base" {
+  # profile       = "terraform-developer"
+
+  # AWS API connection
+  access_key      = var.aws_access_key_id
+  secret_key      = var.aws_secret_access_key
+
+  ami_name      = "${local.image_name}-${var.architecture}-${local.now_unix_timestamp}"
+  instance_type = local.aws_spot_instance_types[var.architecture]
```
Comment (jayfranco999) [inline sources.pkr.hcl:10]:
```suggestion
  instance_type = local.aws_spot_instance_types[var.architecture][0]
```

Comment (jayfranco999) [inline sources.pkr.hcl:10]:
```suggestion
  instance_type = [local.aws_spot_instance_types[var.architecture][0]]
```

File: sources.pkr.hcl [hunk 7]
```diff
@@ -7,7 +7,7 @@ source "amazon-ebs" "base" {
   secret_key      = var.aws_secret_access_key
 
   ami_name      = "${local.image_name}-${var.architecture}-${local.now_unix_timestamp}"
-  instance_type = local.aws_spot_instance_types[var.architecture]
+  instance_type = [local.aws_spot_instance_types[var.architecture][0]]
```
Comment (smerle33) [inline sources.pkr.hcl:10]: it need to be a string !
https://developer.hashicorp.com/packer/integrations/hashicorp/amazon/latest/components/builder/ebs#run-configuration

File: sources.pkr.hcl [hunk 8]
```diff
@@ -1,3 +1,46 @@
...
+source "amazon-ebs" "base" {
+  # profile       = "terraform-developer"
+
+  # AWS API connection
+  access_key      = var.aws_access_key_id
+  secret_key      = var.aws_secret_access_key
+
+  ami_name      = "${local.image_name}-${var.architecture}-${local.now_unix_timestamp}"
+  # instance_type = local.aws_spot_instance_types[var.architecture]
+  instance_type = "t2.xlarge"
```
Comment (smerle33) [inline sources.pkr.hcl:11]: to show you that it work with one instance name

File: variables.pkr.hcl [hunk 9]
```diff
@@ -13,6 +13,19 @@ variable "architecture" {
...
+  default = "us-east-2"
+}
+variable "aws_access_key_id" {
+  type    = string
+  default = env("AWS_ACCESS_KEY_ID")
+}
+variable "aws_secret_access_key" {
+  type    = string
+  default = env("AWS_SECRET_ACCESS_KEY")
+}
```
Comment (jayfranco999) [inline variables.pkr.hcl:27]: Need to remove, first let's read the amazon builder provider documentation.
https://developer.hashicorp.com/packer/integrations/hashicorp/amazon#environment-variables

**Comment by dduportal** [comment]
Closing in favor of #1448  #1447 and #1449

**Pull request closed without merging**